cpp/src/arrow/dataset/dataset.h

```diff
@@ -134,6 +135,9 @@ class ARROW_DS_EXPORT InMemoryFragment : public Fragment {
 
 /// @}
 
+using FragmentPtr = std::shared_ptr<Fragment>;
```

Comment:
```suggestion
using FragmentPtr = std::shared_ptr<Fragment>;
```

I think we generally try and avoid using too many `using` statements like this as it can make the code a bit harder to read for newcomers.  `FragmentGenerator` is probably ok as we have some precedent there.